.github/workflows/_dispatcher.yml

```diff
@@ -91,7 +91,7 @@ jobs:
   build-contributor:
     name: "Build contributor"
     needs: [determine-actions, build-tool-config, build-2204]
-    if: ${{ needs.determine-actions.outputs.run_contrib == '1' }}
+    if: ${{ !cancelled() && needs.determine-actions.outputs.run_contrib == '1' }}

```

Comment:
Strange behavior in github actions: If a `needs` job is skipped, then dependent jobs will be skipped, regardless whether the `if` condition passes. 

Publicly known [workaround](https://github.com/orgs/community/discussions/25224) appears to be to use `always()` in the `if` condition, but then the job can't be cancelled. From my testing it appears that `!cancelled()` is also a workaround but it has the benefit of still being cancellable.